src/trace/dma-trace.c

```diff
@@ -351,6 +351,23 @@ void dma_trace_flush(void *t)
 	dcache_writeback_region((void *)t, size);
 }
 
+void dma_trace_on(void)
+{
+	if (trace_data->enabled)
+		return;
+	trace_data->enabled = 1;
+	schedule_task(&trace_data->dmat_work, DMA_TRACE_PERIOD,
+		      DMA_TRACE_PERIOD);
+}
+
+void dma_trace_off(void)
+{
+	if (!trace_data->enabled)
+		return;
+	schedule_task_cancel(&trace_data->dmat_work);
+	trace_data->enabled = 0;
+}
```

Comment:
Shouldn't you first set enabled = 0 then call schedule_task_cancel?

Reply:
The order of operations is intentional. The `schedule_task_cancel()` does not return any error code to handle atm (imho it should since it may not complete sometimes, but it is for different patch). But if there would be an error (or exception thrown in OOP), then the dma-trace object remains in enabled state until the 'cancel' operation completes, so the state change should happen as a second step I believe.

In case of on() operation, I changed state as a first step just in case the schedule implementation immediately switches the execution to the new task, so that `enabled` potentially queried from another context is correct. If there would be any error from `schedule_task()` to handle, I would revert `enabled` back to 0.

Reply:
I understand. I thought that dmat_work task depends somehow on trace_data->enabled. Anyhow, this is the kind of information I would like to see in the commit message.

Reply:
@dbaluta Yes, I agree, I realized this might be useful information for others and was going to add to the commit message, but found it already merged.